rskj-core/src/main/java/co/rsk/pcc/NativeContract.java

```diff
@@ -0,0 +1,208 @@
+                throw new RuntimeException("RSK Transaction is null");
+            }
+
+            Optional<NativeMethod.WithArguments> methodWithArguments = parseData(data);
+
+            // No function found with the given data? => halt!
+            if (!methodWithArguments.isPresent()) {
+                String errorMessage = String.format("Invalid data given: %s.", Hex.toHexString(data));
+                logger.info(errorMessage);
+                throw new NativeContractIllegalArgumentException(errorMessage);
+            }
+
+            // If this is not a local call, then first check whether the function
+            // allows for non-local calls
+            if (!executionEnvironment.isLocalCall() && methodWithArguments.get().getMethod().onlyAllowsLocalCalls()) {
+                String errorMessage = String.format("Non-local-call to %s. Returning without execution.", methodWithArguments.get().getMethod().getName());
+                logger.info(errorMessage);
+                throw new NativeContractIllegalArgumentException(errorMessage);
+            }
+
+            before();
+
+            Object result;
+            try {
+                result = methodWithArguments.get().execute();
+            } catch (NativeContractIllegalArgumentException ex) {
+                String errorMessage = String.format("Error executing: %s", methodWithArguments.get().getMethod().getName());
+                logger.warn(errorMessage, ex);
+                throw new NativeContractIllegalArgumentException(errorMessage);
+            }
+
+            after();
+
+            // Special cases:
+            // - null => null
+            // - empty Optional<?> => null
+            // - nonempty Optional<?> => encoded ?
+            // Note: this is hacky, but ultimately very short and to the point.
+            if (result == null) {
+                return null;
+            } else if (result.getClass() == Optional.class) {
+                Optional<?> optionalResult = (Optional<?>) result;
+                if (!optionalResult.isPresent()) {
+                    return null;
+                } else {
+                    result = optionalResult.get();
+                }
+            }
+
+            return methodWithArguments.get().getMethod().getFunction().encodeOutputs(result);
+        }
+        catch(Exception ex) {
+            logger.error(ex.getMessage(), ex);
+            panicProcessor.panic("nativecontractexecute", ex.getMessage());
+            throw new RuntimeException(String.format("Exception executing native contract: %s", ex.getMessage()), ex);
+        }
+    }
+
+    private Optional<NativeMethod.WithArguments> parseData(byte[] data) {
+        if (data != null && (data.length >= 1 && data.length <= 3)) {
+            logger.warn("Invalid function signature {}.", Hex.toHexString(data));
+            return Optional.empty();
+        }
+
+        if ((data == null || data.length == 0) && getDefaultMethod().isPresent()) {
```

Comment:
if data is null or empty, and there is no default method, the line 180 would try to access an invalid address in the array causing an exception... I think...

Reply:
good catch! will fix.